packages/react-router-redux/modules/ConnectedRouter.js

```diff
@@ -14,17 +14,65 @@ class ConnectedRouter extends Component {
     store: PropTypes.object
   }
 
+  constructor(props, context) {
+    super(props)
+
+    this.inTimeTravelling = false
+
+    // Subscribe to store changes
+    const store = props.store || context.store
+    this.unsubscribe = store.subscribe(() => {
+      // Extract store's location
+      let pathnameInStore = null
+      let searchInStore = null
+      let hashInStore = null
+
+      if (store.getState().router.location) {
+        const location = store.getState().router.location
+        pathnameInStore = location.pathname
+        searchInStore = location.search
+        hashInStore = location.hash
+      }
+      // Extract history's location
+      const {
+        pathname: pathnameInHistory,
+        search: searchInHistory,
+        hash: hashInHistory,
+      } = props.history.location
+
+      // If we do time travelling, the location in store is changed but location in history is not changed
+      if (pathnameInHistory !== pathnameInStore || searchInHistory !== searchInStore || hashInHistory !== hashInStore) {
+        this.inTimeTravelling = true
+        // Update history's location to match store's location
+        props.history.push({
+          pathname: pathnameInStore,
+          search: searchInStore,
+          hash: hashInStore,
+        })
+      }
+    })
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe()
+  }
+
   render() {
     const { store:propsStore, history, children, ...props } = this.props
     let store = propsStore || this.context.store
 
     return (
       <Router {...props} history={history}>
         <Route render={({ location }) => {
-            store.dispatch({
-              type: LOCATION_CHANGE,
-              payload: location
-            })
+            // Dispatch LOCATION_CHANGE except when we're in time travelling
+            if (!this.inTimeTravelling) {
+              store.dispatch({
```

Comment:
Right. So, my idea is going to refactor it to listen the history in the constructor and dispatch it in the listener instead (actually, the same approach I did [here])(https://github.com/supasate/connected-react-router/blob/12217d1bd481d62153ff729cb2f7be8be9e748b7/src/ConnectedRouter.js#L49).